### PR TITLE
[HW7][JS_5]

### PR DESCRIPTION
**Pull request by JackBlaaack**
Work with methods of array and array

### CODE REVIEW

File: JS_5/task2.1.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    if (abc.includes(el.toLowerCase())) {
+      const count = arrOfChar.filter(
+        (char) => char.toLowerCase() === el.toLowerCase()
+      ).length;
+      return count;
+    } else {
+      return el;
+    }
+  });
+  return result.join("");
```
Comment (VladimirLQA) [inline JS_5/task2.1.js:20]:
```suggestion
  const arrOfChar = str.toLowerCase().split("");
  return arrOfChar.map((el) => {
    if (el.match(/[a-z]/)) {
      const count = arrOfChar.filter((char) => char === el).length;
      return count;
    } else {
      return el;
    }
  }).join('');
```

File: JS_5/task2.2.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// 2. У вас есть массив с ценами товаров в чеке. В консоль нужно вывести сумму всех цен и среднюю цену товара.
+//   Итого: 8495 $, средняя цена товара 700 $ - пример сообщения в консоле.  
+
+  const prices = [64, 7556, 345, 7556, 345, 7556, 345, 7556, 433, 345, 756, 123, 942, 3112, 421, 9341, 1212, 8, 43, 41, 345, 341, 21, 321, 123];
+
+const totalOrder = ((prices) => {
+    const result = prices.reduce((curr, result) => curr + result, 0);
+    return result;
+});
```
Comment (VladimirLQA) [inline JS_5/task2.2.js:9]:
```suggestion
const totalOrder = (prices) => {
    const result = prices.reduce((curr, result) => curr + result, 0);
    return result;
};
```

File: JS_5/task2.2.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+console.log(totalOrder(prices));
+
+const avgOrder = ((prices) => {
+    const avg = totalOrder(prices)/prices.length;
+    return avg;
+});
```
Comment (VladimirLQA) [inline JS_5/task2.2.js:16]:
```suggestion
const avgOrder = (prices) => {
    const avg = totalOrder(prices)/prices.length;
    return avg;
};
```

File: JS_5/task2.3.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+const words = ["Masha", "Tolya", "Jack"];
+
+function countVowels(words) {
+    const match = words.match(/[aeiou]/gi);
+    if (match) {
+        return match.length
+    } else {
+        return 0;
+    }
```
Comment (VladimirLQA) [inline JS_5/task2.3.js:12]:
```suggestion
  return words.match(/[aeiou]/gi).length;
```

File: JS_5/task2.4.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+// 4. У вас есть массив со скобками, предположим [ ‘(‘, ‘)’, ‘(‘, ‘)’, ‘)’], количество элементов и последовательность может быть разной.
+//   Нужно выяснить, у каждой ли скобки есть соответствующая пара (открывающая и закрывающая).
+//   Усложнение: в массиве могут быть вложены еще массивы на разной глубине. (ПОПРОБУЙТЕ МЕТОД .flat(), изучите как он работает с разными глубинами вложенности)
+//   Вернуть нужно всё также есть ли у каждой скобочки соответствующая пара. 
+
+const arr = [[['(']], ')', '(', ')', ')',['(', ['('], [')']]].flat(Infinity);
```
Comment (VladimirLQA) [inline JS_5/task2.4.js:6]: Use `.flat()` inside the function

File: JS_5/task2.4.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+function pairOfBrackets(arr) {
+const result = arr.reduce((balance, el) => {
+if (el === '(') {
+   balance++;
+} else if (el === ')') {
+    balance --;
+} 
+return balance;
+}, 0);
+return result;
```
Comment (VladimirLQA) [inline JS_5/task2.4.js:18]: Return result is a number, but the task is to determine whether the number of open and closed brackets is equal, so  return  should be the boolean value
Same for the name of function, for example, `isEqualBrackets`

**Review by VladimirLQA — DISMISSED** [review]

File: JS_5/task2.4.js (hunk 6 above)
Comment (JackBlaaack) [reply JS_5/task2.4.js:18]: Thanks, I'll fix it.

**Review by VladimirLQA — APPROVED** [review]

**Pull request merged**